myway/src/main/java/com/hmel/myway/dao/blogic/services/BaseHibernateDAO.java

```diff
@@ -51,14 +48,23 @@ public T findOne(P id) throws PhoneDictionaryException {
 
 	}
 
+	/*
+	 * int firstResult, int maxResults
+	 */
 	@SuppressWarnings("unchecked")
 	@Transactional(value = "transactionManager")
-	public List<T> findAll() throws PhoneDictionaryException {
+	public List<T> findAll(int firstResult, int maxResults) throws PhoneDictionaryException, IllegalArgumentException{
 		logger.info("findAll CALLING");
 		getCurrentSession().beginTransaction();
+		if(firstResult<0){
+			throw new IllegalArgumentException("first result must be >=0");
+		}
+		if(maxResults<=0 || maxResults<firstResult){
```

Comment:
I have some confuse with it. In the condition you have OR in the message AND. What it correct?